frontend/src/bundles/common/components/create-goal-form/validation-schemas/goal.validation-schema.ts

```diff
@@ -0,0 +1,34 @@
+import { UserValidationMessage } from 'shared';
+import { z } from 'zod';
+
+import { ActivityType } from '~/bundles/goals/enums/enums.js';
+
+import { GoalFrequency } from '../enums/enums.js';
+
+type GoalRequestValidation = {
+    activity: z.ZodNativeEnum<typeof ActivityType>;
+    frequency: z.ZodNativeEnum<typeof GoalFrequency>;
+    distance: z.ZodOptional<z.ZodString>;
+    duration: z.ZodOptional<z.ZodString>;
+};
+
+const regexDecimalNumbers = new RegExp(/^\d+$/);
+
+const goalValidationSchema = z.object<GoalRequestValidation>({
+    activity: z.nativeEnum(ActivityType),
+    frequency: z.nativeEnum(GoalFrequency),
+    distance: z
```

Comment:
why distance and duration are stored as strings not as numbers?

Reply:
> why distance and duration are stored as strings not as numbers?

I do not know how to get number of input using React-hook-form

Reply:
there should be a way to retrieve value as number, I even think if input has type="number" it will be retrieved as number by default, try to explore docs or we could try to use z.coerce but I don't like this approach

Reply:
input type="number" also returns string